### PR TITLE
 Remove optionalTarget: More clean up

### PR DESCRIPTION
**Pull request by Mugen87**
Follow up #13522

### CODE REVIEW

File: src/objects/Mesh.js [hunk 1]
```diff
@@ -210,7 +212,12 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 				}
 
-				intersection.face = new Face3( a, b, c, Triangle.normal( vA, vB, vC ) );
+				var face = new Face3( a, b, c );
+
+				triangle.set( vA, vB, vC ).normal( faceNormal );
```
Comment (Mugen87) [inline src/objects/Mesh.js]: @WestLangley In context of `BufferGeometry.center()`, the semantics of `Triangle.normal()` is also problematic. It should be `Triangle.getNormal()`...

Comment (WestLangley) [reply src/objects/Mesh.js]: Yes, I guess it should be `Triangle.getNormal( a, b, c, target )` and `Triangle.prototype.getNormal( a, b, c, target )`.

Also `barycoordFromPoint()` -> `getBarycoord()`.

Separate PR, though.

File: src/objects/Mesh.js [hunk 2]
```diff
@@ -210,7 +212,12 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
...
+				face.normal.copy( faceNormal );
```
Comment (WestLangley) [inline src/objects/Mesh.js]: In any event, you can write your existing code like so:
```javascript
triangle.set( vA, vB, vC ).normal( face.normal );
```
Or, better yet:
```javascript
Triangle.normal( vA, vB, vC, face.normal );
```

Comment (Mugen87) [reply src/objects/Mesh.js]: Much better!

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
I've checked all examples, the editor and the docs and found no more warnings so far :+1:

**Comment by WestLangley** [comment]
Thanks, @Mugen87 !

**Pull request merged**